app/Console/Commands/PostQuantity.php

```diff
@@ -0,0 +1,123 @@
+<?php
+
+namespace Rogue\Console\Commands;
+
+use Rogue\Models\Signup;
+use Illuminate\Console\Command;
+
+class PostQuantity extends Command
+{
+    /**
+     * The name and signature of the console command.
+     *
+     * @var string
+     */
+    protected $signature = 'rogue:postquantity {start=1}';
+
+    /**
+     * The console command description.
+     *
+     * @var string
+     */
+    protected $description = 'Add the quantity from the corresponding signup onto the corresponding post.';
+
+    /**
+     * Create a new command instance.
+     *
+     * @return void
+     */
+    public function __construct()
+    {
+        parent::__construct();
+    }
+
+    /**
+     * Execute the console command.
+     *
+     * @return mixed
+     */
+    public function handle()
+    {
+        // Signup that we should start with
+        $start = $this->argument('start');
+
+        // Create the progress bar
+        $bar = $this->output->createProgressBar(Signup::where('id', '>=', $start)->count());
+
+        // Get all signups starting from $start in order of ID
+        Signup::where('id', '>=', $start)->orderBy('id')->with('posts')->chunk(100, function ($signups) use ($bar) {
+            foreach ($signups as $signup) {
+                // Get the posts for the signup
+                $posts = $signup->posts;
+
+                // If no posts, move on
+                if ($posts->isEmpty()) {
+                    info('rogue:postquantity: No posts for signup ' . $signup->id);
+                    $bar->advance();
+                    continue;
+                }
+
+                // Sum quant of posts
+                info('rogue:postquantity: Updating posts for signup ' . $signup->id);
+                $postQuantityTotal = $posts->sum('quantity');
+
+                // If >0, see if any posts have null quant
+                if ($postQuantityTotal) {
+                    $postQuantities = $posts->pluck('quantity')->toArray();
+
+                    // If none have null quant, continue we are done here
+                    if (! in_array(null, $postQuantities, true)) {
+                        continue;
+                    }
+
+                    // If some have null quant we need to fill them in
+                    $missingQuantity = $signup->getQuantity() - $postQuantityTotal;
```

Comment:
Just making sure I got this, we need this logic in there so we can handle posts that come in while this change is being made, right?

Reply:
@sbsmith86 This came out of the impromptu quantity discussion we had about doing the subtraction on the Rogue side. Regardless of what quantity we are returning, if we are doing any storing of quantity on the posts before the script runs, we need this logic to work with signups that have some posts with quantity and some without.

Reply:
right right ok!